text/src/autogluon/text/automm/models/categorical_transformer.py

```diff
@@ -0,0 +1,275 @@
+
+    The following illustration demonstrates one Transformer block for :code:`prenormalization=True`:
+
+    .. image:: ../images/transformer_block.png
+        :scale: 25%
+        :alt: PreNorm Transformer block
+
+    Examples:
+        .. testcode::
+
+            x_num = torch.randn(4, 3)
+            x_cat = torch.tensor([[0, 1], [1, 0], [0, 2], [1, 1]])
+
+            module = FTTransformer.make_baseline(
+                n_num_features=3,
+                cat_cardinalities=[2, 3],
+                d_token=8,
+                n_blocks=2,
+                attention_dropout=0.2,
+                ffn_d_hidden=6,
+                ffn_dropout=0.2,
+                residual_dropout=0.0,
+                d_out=1,
+            )
+            x = module(x_num, x_cat)
+            assert x.shape == (4, 1)
+
+            module = FTTransformer.make_default(
+                n_num_features=3,
+                cat_cardinalities=[2, 3],
+                d_out=1,
+            )
+            x = module(x_num, x_cat)
+            assert x.shape == (4, 1)
+
+        To learn more about the baseline and default parameters:
+
+        .. testcode::
+
+            baseline_parameters = FTTransformer.get_baseline_transformer_subconfig()
+            default_parameters = FTTransformer.get_default_transformer_config()
+
+    References:
+        * [gorishniy2021revisiting] Yury Gorishniy, Ivan Rubachev, Valentin Khrulkov, Artem Babenko, "Revisiting Deep Learning Models for Tabular Data", 2021
+        * [vaswani2017attention] Ashish Vaswani, Noam Shazeer, Niki Parmar, Jakob Uszkoreit, Llion Jones, Aidan N. Gomez, Lukasz Kaiser, Illia Polosukhin, "Attention Is All You Need", 2017
+    """
+
+    def __init__(
+        self, 
+        prefix: str, 
+        num_categories: List[int],
+        d_token: int,
+        # n_tokens: Optional[int] = None,
+        cls_token: Optional[bool] = False,
+        out_features: Optional[int] = None,
+        num_classes: Optional[int] = 0,
+        bias: Optional[bool] = True,
+        initialization: Optional[str] = 'normal',
+        n_blocks: Optional[int] = 2,
+        attention_n_heads: Optional[int] = 8,
+        attention_initialization: Optional[str] = 'kaiming',
+        attention_normalization: Optional[str] = 'LayerNorm',
+        attention_dropout: Optional[str] = 0.2, 
+        residual_dropout: Optional[str] = 0.0,
+        ffn_activation: Optional[str] = 'ReGLU',
+        ffn_normalization: Optional[str] = 'LayerNorm',
+        ffn_d_hidden: Optional[str] = 6,
+        ffn_dropout: Optional[str] = 0.0,
+        prenormalization: Optional[bool] = True,
+        first_prenormalization: Optional[bool] =  False,
+        last_layer_query_idx: Union[None, List[int], slice] = None,
+        kv_compression_ratio: Optional[float] = None,
+        kv_compression_sharing: Optional[str] = None,
+        head_activation: Optional[str] =  'ReLU',
+        head_normalization: Optional[str] = 'LayerNorm',
+    ):
```

Comment:
Consider to add additional docstrings to explain the arguments.

Reply:
Done for both `categorical_transformer` and `numerical_transformer`.